labs/b7.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Lab 7 - Services
+layout: lab
+---
+
+
+## Using systemd
+
+### What services are running right now?
+Run `systemctl`. You’ll see a long table of every unit known to systemd.
+
+Let’s narrow it down to services for now. Run `systemctl --type=service`. Now you can see a list of all services running on your computer. Each of these services is a daemon running in the background. Do you see any familiar services running?
+
+### Controlling Services
+Now let's use `systemd` to control a an nginx web server. Install nginx by issuing `sudo apt install nginx`. Once that is done we can tell systemd to start the service with the following: `sudo systemctl start nginx`. Run `systemctl status nginx` to ensure it is running and navigate to http://*yourvm*.decal.xcf.sh/ -- you should be greeted by the nginx default landing page.
+
+Now let's make nginx listen for connections on the nonstandard port 420. In `/etc/nginx/sites-available/default` change the following lines:
+
+```nginx
+listen 80 default_server;
+listen [::]:80 default_server;
+```
+
+to:
+```nginx
+listen 420 default_server;
+listen [::]:420 default_server;
+```
+
+Tell systemd that nginx has changed configuration and needs reloading with: `sudo systemctl reload nginx`. Now, accessing  http://*yourvm*.decal.xcf.sh/ should now give you a connection refused error and your webserver will only be accessible via http://*yourvm*.decal.xcf.sh:420/.
+
+Note that not all services can be reloaded; systemd will notify you if this is the case and such services will have to be restarted instead with: `sudo systemctl restart yourservice`.
+
+Finally go ahead and stop the nginx service with `sudo systemctl stop nginx`.
+
+(Exercise 1) What is the difference between `systemctl reload yourservice` and `systemctl reload restart yourservice`?
+(Exercise 2) What determines the specific behavior of `systemctl reload yourservice` and `systemctl reload restart yourservice`?
```

Comment:
```suggestion
(Exercise 2) What determines the specific behavior of `systemctl reload yourservice` and `systemctl restart yourservice`?
```

Can you explain this question a bit more? I'm not really sure what you're asking.